src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Hybrid/HybridCache.cs

```diff
@@ -45,6 +46,111 @@ public ValueTask<T> GetOrCreateAsync<T>(string key, Func<CancellationToken, Valu
+    /// <param name="factory">Provides the underlying data service if the data is not available in the cache.</param>
+    /// <param name="state">The state required for <paramref name="factory"/>.</param>
+    /// <param name="options">Additional options for this cache entry.</param>
+    /// <param name="tags">The tags to associate with this cache item.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be canceled.</param>
+    /// <returns>The data, either from cache or the underlying data service.</returns>
+    /// <remarks>Implementors may use the key span to attempt a local-cache synchronous 'get' without requiring the key as a <see cref="string"/>.</remarks>
+    public virtual ValueTask<T> GetOrCreateAsync<TState, T>(
+        ReadOnlySpan<char> key,
+        TState state,
+        Func<TState, CancellationToken, ValueTask<T>> factory,
+        HybridCacheEntryOptions? options = null,
+        IEnumerable<string>? tags = null,
+        CancellationToken cancellationToken = default)
+        => GetOrCreateAsync(key.ToString(), state, factory, options, tags, cancellationToken);
+
+    /// <summary>
+    /// Asynchronously gets the value associated with the key if it exists, or generates a new entry using the provided key and a value from the given factory if the key is not found.
+    /// </summary>
+    /// <typeparam name="T">The type of the data being considered.</typeparam>
+    /// <param name="key">The key of the entry to look for or create.</param>
+    /// <param name="factory">Provides the underlying data service if the data is not available in the cache.</param>
+    /// <param name="options">Additional options for this cache entry.</param>
+    /// <param name="tags">The tags to associate with this cache item.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be canceled.</param>
+    /// <returns>The data, either from cache or the underlying data service.</returns>
+    public ValueTask<T> GetOrCreateAsync<T>(
+        ref DefaultInterpolatedStringHandler key,
+        Func<CancellationToken, ValueTask<T>> factory,
+        HybridCacheEntryOptions? options = null,
+        IEnumerable<string>? tags = null,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            return GetOrCreateAsync(key.Text, factory, WrappedCallbackCache<T>.Instance, options, tags, cancellationToken);
+        }
+        finally
+        {
+            // It is *not* an error that this Clear occurs before the "await"; by definition, the implementation is *required* to copy
+            // the value locally before an await, precisely because the ref-struct cannot bridge an await. Thus: we are fine to clean
+            // the buffer even in the non-synchronous completion scenario.
+            key.Clear();
+        }
```

Comment:
```suggestion
        ValueTask<T> result = GetOrCreateAsync(key.Text, factory, WrappedCallbackCache<T>.Instance, options, tags, cancellationToken);
        key.Clear();
        return result;
```

In other places we do not return pooled arrays (which is what `Clear` might do) in a `finally`.